### PR TITLE
PROD-744 Fix 2903 Issue with youtube preview whenever link has space at the end

### LINKED ISSUE #2903: Issue with youtube video preview whenever link has space at the end
**Describe the bug**
Issue happens when you paste a youtube link with space at the end. Preview becomes different compared to normal preview of youtube link.

**To Reproduce**
1. Copy youtube link and add space at the end.
2. Paste it in post update.
3. You'll  see that the preview is not normal.

**Expected behavior**
Youtube link preview should always work as what it is normally.

**Screenshots**

**Environment**
Which site did you replicate this issue on?
Demo site

**Support ticket links**
https://secure.helpscout.net/conversation/1616792985/161515?folderId=4312419

**Jira issue** : [PROD-744]

[PROD-744]: https://buddyboss.atlassian.net/browse/PROD-744?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### PR DESCRIPTION
**Pull request by bb-yudhisthir**
Fixed: preview becomes different compared to normal URL embed of YouTube link.

### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

Fixes #2903.

### How to test the changes in this Pull Request:

1. Paste the YouTube share link and
2. Remove old preview, if already
3. Type space at end of the YouTube share link
4. Preview should appear earlier it was not working due to &nbsp; at the end of the link

### Proof Screenshots or Video

http://somup.com/crQuiBrPjY

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Removed the &nbsp; from the URL. Preview should appear now earlier it was not working due to &nbsp; at the end of the link

### CODE REVIEW

File: src/bp-templates/bp-nouveau/js/buddypress-activity-post-form.js [hunk 1]
```diff
@@ -1937,6 +1937,7 @@ window.bp = window.bp || {};
 
 			getURL: function ( prefix, urlText ) {
 				var urlString   = '';
+				urlText         = urlText.replace(/&nbsp;/g, '');
```
Comment (dipskakadiya) [inline src/bp-templates/bp-nouveau/js/buddypress-activity-post-form.js:1940]: @bb-yudhisthir I think `trim` is the proper solution than `replace`. 

```Trim() removes all whitespace characters from the beginning and end of the string. That means spaces , tabs , new lines , returns , and other assorted whitespace characters. Replace() only replaces the designated characters with the given replacement.```

**Review by dipskakadiya — CHANGES_REQUESTED** [review]

File: src/bp-templates/bp-nouveau/js/buddypress-activity-post-form.js (hunk 1 above)
Comment (bb-yudhisthir) [reply src/bp-templates/bp-nouveau/js/buddypress-activity-post-form.js:1940]: trim() replaces the whitespace but doesn't replace the "&nbsp;" chars, I have already tested it http://somup.com/crQUfPr9RA
Also we are already using the trim in subsequent function - loadURLPreview() but not working for "&nbsp" chars

**Review by dipskakadiya — DISMISSED** [review]

**Comment by basil-buddyboss** [comment]
Retested in the QA ongoing branch.
Status: Not Fixed

Proof Screenshots or Video:
https://watch.screencastify.com/v/Fwtu6KAdx3vIySzzyN76

**Review by jitendrabanjara1991 — APPROVED** [review]
LGTM

**Pull request merged**